phonon/client.py

```diff
@@ -0,0 +1,42 @@
+import redis
+import zlib
+
+
+class ShardedClient(object):
+
+    key_mappings = {
+        'get': 0, 'hget': 0,
+        'set': 0, 'hset': 0,
+        'hgetall': 0,
+    }
+
+    def __init__(self, hosts=None, port=6379, db=0):
+        self.hosts = sorted(hosts)
+        self.clients = [redis.StrictRedis(host=host, port=port, db=db) for host in self.hosts]
```

Comment:
We rely on the StrictRedis client for connection pooling. Failover is not handled beyond what redis gives out outta the box in this basic implementation. In the case of an unrecoverable node we'll lose that fraction of the dataset. We'd have to restore an image with the same hostname to ensure consistent routing. Those types of failures should be handled by the application in the form of TTLs on the orphaned keys.